packages/block-editor/src/hooks/style.js

```diff
@@ -323,28 +305,52 @@ const withElementsStyles = createHigherOrderComponent(
 			'link'
 		);
 
-		// The Elements API only supports link colors for now,
-		// hence the specific omission of `link` in the elements styles.
-		// This might need to be refactored or removed if the Elements API
-		// changes or `link` supports styles beyond `color`.
-		const elements = skipLinkColorSerialization
-			? omit( props.attributes.style?.elements, [ 'link' ] )
-			: props.attributes.style?.elements;
-
-		const styles = compileElementsStyles(
-			blockElementsContainerIdentifier,
-			elements
-		);
+		let styles = [];
+
+		// Remove values based on whether serialization has been skipped for a specific style.
+		const rawElementsStyles = props.attributes.style?.elements;
+
+		if ( ! isEmpty( rawElementsStyles ) ) {
+			const filteredElementsStyles = {
+				...rawElementsStyles,
+				link: {
+					...rawElementsStyles?.link,
+					color: ! skipLinkColorSerialization
+						? rawElementsStyles?.link?.color
+						: undefined,
+				},
+			};
+
+			styles = Object.entries( filteredElementsStyles ).reduce(
+				( acc, [ elementName, elementStyles ] ) => {
+					const cssRule = generateStyles( elementStyles, {
+						// The .editor-styles-wrapper selector is required on elements styles. As it is
+						// added to all other editor styles, not providing it causes reset and global
+						// styles to override element styles because of higher specificity.
+						selector: `.editor-styles-wrapper .${ blockElementsContainerIdentifier } ${ ELEMENTS[ elementName ] }`,
+					} );
+
+					if ( !! cssRule ) {
+						acc.push( cssRule );
+					}
+
+					return acc;
+				},
+				[]
+			);
+		}
+
 		const element = useContext( BlockList.__unstableElementContext );
+		const hasElementStyles = styles.length;
 
 		return (
 			<>
-				{ elements &&
+				{ hasElementStyles &&
 					element &&
 					createPortal(
 						<style
 							dangerouslySetInnerHTML={ {
-								__html: styles,
+								__html: styles.join( ' ' ),
```

Comment:
Any reason why we're joining them using an empty space rather than a line break (`\n`)?

Reply:
No reason. Just a personal preference. 

I've added the line break back in. It was there before. Good catch.